src/Raven.Server/Storage/Schema/Updates/Documents/50000/From42017.cs

```diff
@@ -141,6 +139,86 @@ public bool Update(UpdateStep step)
             return true;
         }
 
+        private static unsafe void UpdateDocumentCounters(UpdateStep step, DocumentsOperationContext context, string docId)
+        {
+            using (DocumentIdWorker.GetSliceFromId(context, docId, out Slice lowerDocId))
+            {
+                var docsTable = new Table(DocsSchema, step.ReadTx);
+                if (docsTable.ReadByKey(lowerDocId, out var tvr) == false) 
+                    return; // document doesn't exists 
+                
+                var tableId = tvr.Id;
+                var counterNames = step.DocumentsStorage.CountersStorage.GetCountersForDocumentList(context, step.WriteTx, docId);
```

Comment:
better to use `GetCountersForDocument()` and do `.ToList()` 

This way we can avoid having `GetCountersForDocumentList` method